### ISSUE TITLE
export with python function: Kcov get hangs

### ISSUE THREAD

**Issue by bhaskarrana** [SimonKagstrom/kcov#188]
Hi Simon,

As below is the script where we call python  by exporting its python function
cat test.sh 

#!/bin/bash -x
function python(){
/usr/bin/kcov  /tmp/cov $@
}
export -f python
python test.py

cat test.py

#!/usr/bin/python
print  "Calling the python"

We are getting following error

kcov: error: Can't create python FIFO /tmp/cov//python-helper.py/kcov-python.pipe

kcov: error: Can't start/attach to /tmp/cov/python-helper.py

and program get hangs .

To get rid of this error .
Made the following changes in the respective file **src/configuration.cc** of kcov source code .
setKey("python-command", **"`which python`");**
as earlier it was
setKey("python-command", **"python");**

After making subsequent changes , kcov does not hangs and we are getting proper python coverage .
Kindly suggest , is this a correct way  .

Regards
Bhaskar Rana

**Comment by SimonKagstrom** [comment]
I'm OK with that change, can you prepare a pull request?

Another possibility is to use `kcov --python-parser=<your-wrapper>` in the call to kcov, but your solution should work better out of the box.

Thanks!

**Comment by SimonKagstrom** [comment]
I tried this fix, but unfortunately it doesn't work for me. I just get

```
/usr/bin/python
which: no python-helper.py in (/tmp/kcov)
/home/ska/projects/kcov/tests/python/main
```

**Comment by bhaskarrana** [comment]
Hi Simon,
I am doing following things for pull request
git clone https://github.com/SimonKagstrom/kcov.git
cd kcov/
git checkout -b pythonquotes
vi src/configuration.cc
git status
git add  src/configuration.cc
git config --global user.name "bhaskar rana"
git config --global user.email bhaskar_rana30@yahoo.com
git commit -m "Fixing the python quotes issue 183"
git push origin pythonquote


Yields following error

 git push origin pythonquote
error: The requested URL returned error: 403 Forbidden while accessing https://github.com/SimonKagstrom/kcov.git/info/refs

I do not what wrong i am doing here  or some permission issue.
By bad it is not tilde but **backtick** , which  will be in the begining and end for **which python**  command.
http://unix.stackexchange.com/questions/27428/what-does-backquote-backtick-mean-in-commands
That will resolve the problem of hanging .

Regards
Bhaskar Rana

**Comment by SimonKagstrom** [comment]
Yes, you're trying to push to my git repo, which isn't allowed. The "github" way is to fork the repo, make changes there and then send a pull request. You can see a guide here:

https://guides.github.com/activities/forking/

I can certainly do the commit myself, but since you're the author of it, it would be good if that shows up in the commit history.


Anyway, thanks for the "ticks" :-), didn't notice those!

A certain complication with that solution is that it will be more UNIX-specific. kcov doesn't run on Windows today, but if issue #185 is solved, it would be technically possible. Windows probably doesn't like ticks as far as I know.

**Comment by SimonKagstrom** [comment]
Hi again Bhaskar! Could you prepare a pull request for this fix?

**Comment by bhaskarrana** [comment]
Hi Simon,
Sorry for delayed response , probably will do it this week.
RegardsBhaskar Rana 

    On Monday, 9 October 2017 9:20 AM, Simon Kagstrom <notifications@github.com> wrote:
 

 Hi again Bhaskar! Could you prepare a pull request for this fix?—
